sdk/python/feast/feature_store.py

```diff
@@ -318,10 +318,11 @@ def _get_features(
 
         _feature_refs: List[str]
         if isinstance(_features, FeatureService):
-            # Get the latest value of the feature service, in case the object passed in has been updated underneath us.
-            _feature_refs = _get_feature_refs_from_feature_services(
-                self.get_feature_service(_features.name)
-            )
+            _feature_refs = self.get_feature_service(_features.name).features
+            if _features.features != _feature_refs:
+                raise ValueError(
+                    "FeatureService object that's passed in is inconsistent with version from Registry."
```

Comment:
I am not sure if raising an exception here is the right approach. We are getting the latest version back. The reason we query the registry is because it may have a newer version. If we go with this approach of raising an exception, we might run into a case where one person keeps updating the registry and another person keeps getting exceptions. 

I think it's sufficient to print a warning.

Also, we can be clearer than "FeatureService object that's passed in is inconsistent with version from Registry.". What about "A newer version of the FeatureService \"{name}"\ has been found.". What do you think?

Reply:
sounds good. I'm doing this - 

```
                warnings.warn(
                    "The FeatureService object that has been passed in as an argument is"
                    "inconsistent with the version from Registry. Potentially a newer version"
                    "of the FeatureService has been applied to the registry.")
```